lib/cog_api/http/roles.ex

```diff
@@ -7,8 +7,8 @@ defmodule CogApi.HTTP.Roles do
     Base.get(endpoint, "roles") |> Base.format_response("roles", [%Role{}])
   end
 
-  def show(%Endpoint{}=endpoint, id) do
-    Base.get(endpoint, "roles/#{id}") |> Base.format_response("role", %Role{})
+  def show(%Endpoint{}=endpoint, role_name) do
+    Base.get_by(endpoint, "roles", name: role_name) |> Base.format_response("role", %Role{})
   end
```

Comment:
If you will already have an id that you can send, then this can stay as is. It was my understanding that you did not have the role ID at this point.